src/test/java/com/zaxxer/nuprocess/RunTest.java

```diff
@@ -0,0 +1,468 @@
+/*
+ * Copyright (C) 2019 Brett Wooldridge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.zaxxer.nuprocess;
+
+import com.zaxxer.nuprocess.codec.NuAbstractCharsetHandler;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.CoderResult;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.zip.Adler32;
+
+/**
+ * Performs <i>synchronous</i> tests using {@link NuProcessBuilder#run}.
+ */
+public class RunTest
```

Comment:
This file is essentially a copy of `CatTest`, but it uses `run` instead of `start`. The goal here is to apply all the same validations that are applied for asynchronous execution to synchronous execution. (This test turned up the `onExit` issue on Linux, for example)